src/views/Home.jsx

```diff
@@ -1,17 +1,55 @@
 import './Home.css';
 import { Navigate } from 'react-router-dom';
-import { useCallback } from 'react';
+import { useCallback, useState } from 'react';
+import { verifyExistingList } from '../api/firebase';
+
+export function Home({ createNewList, listToken, setListToken }) {
+	const [token, setToken] = useState('');
 
-export function Home({ createNewList, listToken }) {
 	const handleClick = useCallback(() => {
 		createNewList();
 	}, [createNewList]);
 
+	const formHandler = (e) => {
+		const value = e.target.value;
+		setToken((token) => value);
+	};
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+		const formattedToken = token.toLowerCase();
+		const tokenExists = await verifyExistingList(formattedToken);
+
+		if (tokenExists) {
+			//navigate to list if token exists
+			setListToken(formattedToken);
+		} else {
+			//invalid token alert or error message
+			alert('This list does not exist.');
```

Comment:
Good feedback for the user. Another place where we can eventually use a global alert component.

Reply:
➕